cmake/nanobind-config.cmake

```
@@ -120,6 +120,7 @@ function (nanobind_build_library TARGET_NAME)
     ${NB_DIR}/src/trampoline.cpp
     ${NB_DIR}/src/implicit.cpp
   )
+  add_library(nanobind::nanobind ALIAS ${TARGET_NAME})
```

Do you mean the following?
```suggestion
  add_library(nanobind::${TARGET_NAME} ALIAS ${TARGET_NAME})
```

---

I'm skeptical of the utility of this PR. I don't know see the advantage of this ALIAS target if nanobind is being used as documented. But it definitely seems inappropriate to erase the abstraction provided by this function by aliasing all TARGET_NAMEs to the same alias, even if only one target name is ever used.

If it is merely for the convenience of the consumer who wants to use the `nanobind*` targets directly, instead of through `nanobind_add_module`, then such a consumer could make the alias in their own project.

In the example linked by @friendlyanon , it looks like this is, in fact, intended to help with (undocumented) direct use of the `nanobind` target by the consumer build system, for which a statically named alias for the target would be useful. In that case, this alias target would be more appropriately declared at, say, line 274.

But, again, I don't think this is a use case that you, @wjakob , intend to support this way. If the consuming project can't find or set the details it needs through the CMake input variables and `nanobind_add_module()` It seems like what may actually be needed is some more output variables from the config module or PARENT variables from the CMake subdirectory.

In any case, after reviewing the linked example, I don't think my proposed change is what the author intended, so I'll withdraw the suggestion.

---

Oh, you're right. Sorry @wjakob . After the third time through, I finally saw the sentence in https://github.com/wjakob/nanobind/pull/201#issuecomment-1529898556 that makes my comment completely pointless. I'll delete.